### ISSUE TITLE
How to get the ID of the sent message?

### ISSUE THREAD

**Issue by dpominov** [Rhymen/go-whatsapp#43]
To know to which message the confirmation (Ack) came I need to know the ID of the sent message. How to get the ID when sending a message?

**Comment by SchulteMK** [comment]
If the ID is not set or shorter than 2 characters the ID gets generated automatically. At the moment there is no way for you to receive the generated ID. But you could generate the ID for yourself and pass it with the message and keep track of it.
[Our code for generating the message IDs:](https://github.com/Rhymen/go-whatsapp/blob/92ffb4c11a14df04cc0fd8b285b8ef2efb023d54/message.go#L138)
```
b := make([]byte, 10)
rand.Read(b)
info.Id = strings.ToUpper(hex.EncodeToString(b))
```

**Comment by SchulteMK** [comment]
I don´t think it is necessary to have a public method. By testing we found out that the ID doesn´t follow strict requirements. It has to be at least 2 characters long and that´s it. You can use any kind of random text. I´m not totally sure about the maxiumum length though.
We just did it this way, because it is the exact same way as WhatsApp does it.

**Comment by KrzysztofMadejski** [comment]
As there are 3 issues regarding this problem and I just wanted to add 4th one it would be good to document that in the README and examples.

---

Another option would be to pass it back:
1. As a return value
2. Pass msg argument by reference, deep into getInfoProto you set the `msg.Info.Id` but it is not returned back

But both changes are breaking.
